src/encryption/symmetric/aes/mod.rs

```diff
@@ -109,6 +136,31 @@ pub struct AES<const N: usize> {}
 #[derive(Debug, Default, Clone, Copy, PartialEq)]
 struct State([[u8; 4]; 4]);
 
+/// Multiplies a 8-bit number in the Galois field GF(2^8). This is done using "carry-less
+/// multiplication" or "bitwise multiplication", where the binary representation of each
+/// element is treated as a polynomial.
+///
+/// NOTE: this multiplication is not commutative - ie. A * B may not equal B * A.
+fn galois_multiplication(mut col: u8, mut multiplicant: usize) -> u8 {
```

Comment:
just want to get others' opinions, is it possible to use $GF(2^{8})$ throughout, rather than have these specific implementations? 

I think this will make traversing the code a bit simpler, but might make other operations like `SubBytes`/`AddRoundKeys` more difficult.

Reply:
Yeah that's a good point, in some parts of the code (such as this) we are indeed operating in $GF(2^8)$. I could see this change being easier to understand/read (since we already have `GaloisField` implemented), and that should really be the main goal.

Reply:
I think this would be a good idea. It would be nice to stand on the other primitives we have so we don't have to reinvent them every time